src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -2364,7 +2367,55 @@ void M([|D|]&lt;int&gt; lambda)
 
             var navigationSymbol = await context.GetNavigationSymbolAsync();
             var metadataAsSourceFile = await context.GenerateSourceAsync(navigationSymbol);
-            context.VerifyResult(metadataAsSourceFile, expected);
+            context.VerifyResult(metadataAsSourceFile, new Lazy<string>(() => expected));
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        [InlineData(LanguageNames.CSharp)]
+        [InlineData(LanguageNames.VisualBasic)]
+        public async Task TestFailedDecompilation(string languageName)
+        {
+            var exportProvider = ExportProviderCache
+                .GetOrCreateExportProviderFactory(
+                    TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(typeof(ThrowingDecompiledSourceService)))
+                .CreateExportProvider();
+            var throwingService = exportProvider.GetExportedValue<ThrowingDecompiledSourceService>();
+            var expectedException = throwingService.Exception;
```

Comment:
Inline expectedException? That way you're not going to risk use of the unthrown exception?